### ISSUE TITLE
Bound volumes must be repeated

### ISSUE THREAD

**Issue by gotgenes** [docker/docker-py#800]
From the [less verbose example of mounting volumes](http://docker-py.readthedocs.org/en/latest/volumes/),

``` python
container_id = c.create_container(
    'busybox', 'ls', volumes=['/mnt/vol1', '/mnt/vol2'],
    host_config=docker.utils.create_host_config(binds=[
        '/home/user1/:/mnt/vol2',
        '/var/www:/mnt/vol1:ro',
    ])
)
```

we can see that `'/mnt/vol1'` and `'/mnt/vol2'` must be repeated. A preferred API would simply be

``` python
container_id = c.create_container(
    'busybox', 'ls',
    volumes=['/var/www:/mnt/vol1:ro', '/home/user1/:/mnt/vol2']
)
```

In this API, `volumes`
1. is actually a useful parameter when one needs to explicitly bind, without the verbosity of `host_config` and `binds`;
2. mirrors precisely the [the API to `docker run` for mounting volumes](https://docs.docker.com/userguide/dockervolumes/#mount-a-host-directory-as-a-data-volume), so that the user can re-apply the paradigm she has already learned; and
3. removes the need for redundantly specifying the same volume.

Additionally, `volumes` could support the more explicit dictionary argument, similar to `binds`:

``` python
container_id = c.create_container(
    'busybox', 'ls',
    volumes={
        '/home/user1/': {
            'bind': '/mnt/vol2',
            'mode': 'rw',
        },
        '/var/www': {
            'bind': '/mnt/vol1',
            'mode': 'ro',
        }
    }
)
```

**Comment by shin-** [comment]
We wouldn't do that in `create_container` because we want to stick as close as possible to the API itself, which defines those two things separately (config part declares the volumes themselves, host_config part optionally defines the host folder mappings).

However, It could be implemented as a helper, or a high-level wrapper on `create_container`.

**Comment by bfirsh** [comment]
Fixed in #1186.

**Comment by gotgenes** [comment]
@bfirsh Thanks for the update. I couldn't find any comments on #1186 regarding the API change directly addressing this issue. Would you be able to provide more information?

**Comment by bfirsh** [comment]
Ah, sorry. It's not actually released yet with documentation, but in 2.0 you'll be able to do things along the lines of: `client.containers.run("busybox", volumes=['/var/www:/mnt/vol1:ro', '/home/user1/:/mnt/vol2'])`. Ditto `create()`.

**Comment by gotgenes** [comment]
@bfirsh Great! Thanks for following up!
